File: src/main/java/komu/demodel/utils/FileSet.java
/*
 * Copyright (C) 2006 Juha Komulainen. All rights reserved.
 */
package komu.demodel.utils;

import java.io.File;
import java.io.FileFilter;
import java.util.Iterator;

public final class FileSet implements Iterable<File> {

    private final File rootDirectory;
    private final FileFilter filter;
    
    public FileSet(File rootDirectory, FileFilter filter) {
        this.rootDirectory = rootDirectory;
        this.filter = filter;
    }

    public Iterator<File> iterator() {
        return new FileIterator(rootDirectory, filter);
    }
}

File: src/main/scala/komu/demodel/utils/FileSet.scala
/*
 * Copyright (C) 2006-2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.utils

import java.io.{ File, FileFilter }

final class FileSet(rootDirectory: File, filter: FileFilter) extends Iterable[File] {
  assert(rootDirectory != null)
  assert(filter != null)

  def iterator = new FileIterator(rootDirectory, filter)
}
